tests/mve/gtest_scene.cc

```diff
@@ -0,0 +1,715 @@
+#include <gtest/gtest.h>
+
+#include <cstdio>
+#include <iomanip>
+#include <sstream>
+#include <string>
+
+#include "mve/bundle_io.h"
+#include "mve/scene.h"
+#include "util/exception.h"
+#include "util/file_system.h"
+
+namespace {
+
+/* Cleanup mechanism. Unlinks files on destruction. */
+class OnScopeExit final
+{
+public:
+    ~OnScopeExit();
+    void unlink(const std::string& str);
+
+private:
+    void unlink_recursive(const std::string& str);
+
+private:
+    std::vector<std::string> paths;
+};
+
+/**
+ * Creates a valid scene directory in the system's temporary directory.
+ * This directory has an empty "views" subdirectory and
+ * a bundle file with zero cameras and zero features.
+ * @param view_count Number of views the scene is created with.
+ * @param bundle Bundle to include within the scene or
+ *     nullptr to create a scene without a bundle file.
+ * @return The path to the created scene directory.
+ */
+std::string create_scene_on_disk (const std::size_t view_count,
+    const mve::Bundle::Ptr bundle, OnScopeExit& clean_up);
```

Comment:
Regarding function arguments: Either use const references, or pointers if the argument is modified. Let the types speak for themselves :)